### PR TITLE
Currencies: UI + LOGIC

### PR DESCRIPTION
**Pull request by mikolajmirko**
~~Waiting for new snackbars from #45~~

### CODE REVIEW

**Comment by mikolajmirko** [comment]
NOTE:
There is a bug with loading currencies from user in task and reward forms.

File: i18n/pl_PL.json [hunk 1]
```diff
@@ -421,6 +422,25 @@
...
+				"header": {
+					"title": "Waluty"
+				},
+				"content": {
+					"generalHint": "Możesz ustawić do trzech dodatkowych walut w aplikacji, a następnie używać ich w zadaniach i nagrodach. Domyślna waluta aplikacji, nie może zostać zmodyfikowana.",
+					"currenciesUpdatedText": "Zaktualizowano listę walut",
+					"saveCurrenciesButton": "Zapisz konfigurację",
+					"currencyNotUsed": "Nie używana",
+					"currencyNameLabel": "Nazwa waluty",
+					"clearCurrencyText": "Zdobyte w tej walucie punkty nie zostaną usunięte. Nie będzie można jednak używać dalej tej waluty. Na pewno chcesz wyczyścić tą walutę?",
```
Comment (stasgora) [inline i18n/pl_PL.json:436]:
```suggestion
					"clearCurrencyText": "Stracisz możliwość używania tej waluty. Zdobyte już przez dzieci punkty zostaną zachowane. Na pewno chcesz usunąć tą walutę?",
```

File: lib/app.dart [hunk 2]
```diff
@@ -136,6 +138,7 @@ class FokusApp extends StatelessWidget {
 			AppPage.caregiverBadgeForm.name: (context) => _createPage(CaregiverBadgeFormPage(), context, BadgeFormCubit(getParams(context), getActiveUser(context))),
 			AppPage.caregiverStatistics.name: (context) => _createPage(CaregiverStatisticsPage(), context),
 			AppPage.caregiverRatingPage.name: (context) => _createPage(CaregiverRatingPage(), context),
+			AppPage.caregiverCurrencies.name: (context) => _createPage(CaregiverCurrenciesPage(), context, CaregiverCurrenciesCubit(getActiveUser(context), getActiveUser(context), context.bloc<AuthenticationBloc>())),
```
Comment (stasgora) [inline lib/app.dart:141]:
```suggestion
			AppPage.caregiverCurrencies.name: (context) => _createPage(CaregiverCurrenciesPage(), context, CaregiverCurrenciesCubit(getActiveUser(context), getActiveUser(context), authBloc(context))),
```

**Review by stasgora — CHANGES_REQUESTED** [review]
Some text suggestions

File: i18n/en_US.json [hunk 3]
```diff
@@ -421,6 +422,25 @@
...
+			"currencies": {
+				"header": {
+					"title": "Currencies"
+				},
+				"content": {
+					"generalHint": "You can configure up to three different additional currencies for use in tasks and rewards. App's default currency, can't be changed.",
+					"currenciesUpdatedText": "Currencies list has been updated",
+					"saveCurrenciesButton": "Save currencies",
+					"currencyNotUsed": "Not used",
+					"currencyNameLabel": "Nazwa waluty",
```
[the remark this answers is not in the document]
Comment (mikolajmirko) [reply i18n/en_US.json:435]: Ops 🙄

File: i18n/en_US.json [hunk 4]
```diff
@@ -421,6 +422,25 @@
...
+					"clearCurrencyText": "Points in this currency will not be removed. You will however loose the ability to use it. Are you sure you want to clear this currency?",
```
[the remark this answers is not in the document]
Comment (mikolajmirko) [reply i18n/en_US.json:436]: The first part is great, but as like with the icon I intentionally put "clear" there to indicate that you are not really removing/deleting it, just "not using it from now on"... if that makes sense.

[the remark this answers is not in the document]
Comment (stasgora) [reply i18n/en_US.json:436]: Clear could be understood as _only remove the name_

[the remark this answers is not in the document]
Comment (stasgora) [reply i18n/en_US.json:436]: Maybe the first sentence will make it clear..

File: i18n/pl_PL.json (hunk 1 above)
Comment (stasgora) [reply i18n/pl_PL.json:436]: Edited both

File: i18n/en_US.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (mikolajmirko) [reply i18n/en_US.json:436]: Yeah, probably it will. If it will turn out be a problem for users then we will change it to "remove", alright?

**Review by stasgora — APPROVED** [review]

**Pull request merged**